src/esm_parser/esm_parser.py

```diff
@@ -1943,7 +1954,7 @@ def find_variable(tree, rhs, full_config, white_or_black_list, isblacklist):
                         isblacklist,
                     )
 
-                if "$((" in var_result:
+                if isinstance(var_result, str) and "$((" in var_result:
```

Comment:
Do we really need to check the `str` instance here? It is already inside the `if type(var_result) == str:` block

Reply:
Yes, `var_result` that is a boolean, integer or float some times reaches this line, resulting in a crash.

Reply:
Ah, I see that it is returned again inside the if block. Sorry